Makefile

```diff
@@ -145,6 +145,14 @@ runner-build:
 		-t $(RUNNER_IMAGE):$(RUNNER_IMAGE_TAG_PREFIX)builder -f test/e2e/Dockerfile test/e2e
 
 
+	@echo "building target tests-1.24"
+	@docker build --target tests-1.24 \
+		--cache-from $(RUNNER_IMAGE):$(RUNNER_IMAGE_TAG_PREFIX)builder \
+		--cache-from $(CANONICAL_RUNNER_IMAGE):builder \
+		--cache-from $(RUNNER_IMAGE):$(RUNNER_IMAGE_TAG_PREFIX)tests-1.24 \
```

Comment:
You need to include 1.24 in the targets below as well for caching purposes, similar to how, say, 1.23 is used multiple times.